Dockerfile

```diff
@@ -1,7 +1,13 @@
-FROM python:3.11-slim-bookworm
-RUN apt -y update && apt -y upgrade && apt -y install make git
-COPY . /app
+FROM cgr.dev/chainguard/wolfi-base
+ARG python_version=3.11
+
+USER root
+RUN apk update && apk add python3=~${python_version} make git
+
+COPY --chown=nonroot:nonroot . /app
+
+USER nonroot
 WORKDIR /app
-RUN make clean install
-RUN ln -s .venv/bin /app/bin
+RUN make clean install && ln -s .venv/bin /app/bin
+
```

Comment:
I had [this line](https://github.com/elastic/connectors/pull/3041/files#diff-dd2c0eb6ea5cfc6c4bd4eac30934e2d5746747af48fef6da689e85b752f39557R15)
> `RUN apk del make git`

so that we could remove now-unnecessary packages from the final images, with the assumption that we do not need `make` or `git` at runtime, but we could confirm with someone from @elastic/search-extract-and-transform that it makes sense

Reply:
@acrewdson My thought process for omitting that line was: 

- since we're going to scan the built image we'd want to be able to detect vulnerabilities on these packages, even if they're not required at runtime. 
- they've been included in the Dockerfile so removing them now can cause a "breaking" change for users who already build on top of this dockerfile. Changing the base image itself can be a much more substantial breaking change though so perhaps that shouldn't be factored in here.

Let me know what you think?